pep-0654.rst

```diff
@@ -23,49 +25,52 @@ Motivation
 ==========
 
 The interpreter is currently able to propagate at most one exception at a
-time. The chaining features introduced in
-`PEP 3134 <https://www.python.org/dev/peps/pep-3134/>`_ link together
+time. The chaining features introduced in :pep:`3134` link together
 exceptions that are related to each other as the cause or context, but
 there are situations where multiple unrelated exceptions need to be propagated
 together as the stack unwinds. Several real world use cases are listed below.
 
 * **Concurrent errors**. Libraries for async concurrency provide APIs to invoke
   multiple tasks and return their results in aggregate. There isn't currently
   a good way for such libraries to handle situations where multiple tasks
-  raise exceptions. The Python standard library's
-  `asyncio.gather() <https://docs.python.org/3/library/asyncio-task.html#asyncio.gather>`_
+  raise exceptions. The Python standard library's ``asyncio.gather()`` [1]_
   function provides two options: raise the first exception, or return the
-  exceptions in the results list.  The `Trio <https://trio.readthedocs.io/en/stable/>`_
+  exceptions in the results list.  The Trio [2]_
   library has a ``MultiError`` exception type which it raises to report a
   collection of errors. Work on this PEP was initially motivated by the
   difficulties in handling ``MultiErrors``, which are detailed in a design
-  document for an improved version, `MultiError2 <https://github.com/python-trio/trio/issues/611>`_.
+  document for an improved version, ``MultiError2`` [3]_.
   That document demonstrates how difficult it is to create an effective API
   for reporting and handling multiple errors without the language changes we
-  are proposing.
+  are proposing (see also the `Programming Without except*`_ section.)
+
+* **Implementing better Task spawning API in asyncio**.
+  The current asyncio API for spawning child Tasks -- the ``asyncio.gather()``
+  function [1]_ -- has a few shortcomings, the most notable of which is
+  how it handles exceptions.  A better API, inspired by the Trio's "Nursery"
+  primitive [13]_, has been a long-awaited feature of asyncio, blocked on
+  Python not having native language level support for exception groups.
```

Comment:
This needs some more elaboration. Maybe just reference the complexity of Trio's solution?

Reply:
@iritkatriel @gvanrossum I can add a section to the PEP about TaskGroups API. Or you guys think that creating a bpo with the proposed design would be a better move?

Reply:
Adding that design here means we are asking the steering council to approve it as part of this PEP, and if it's not straightforward I think it makes this PEP too big in scope. To Guido's point - isn't this really the same problem/use case as Trio's?

Reply:
Yeah, I think we could shorten this parargaph so it just says "the same applies for asyncio".

Reply:
My problem with removing every notion of TaskGroups from this PEP weakens its motivation.  Ultimately adding task groups to asyncio were the main reason why we started working on this PEP.

Reply:
So say exactly that.

Reply:
I don't think you should remove it, just roll it into the same paragraph with Trio because it's the same kind of use case.  Like the paragraph about multiple use callbacks has two different use cases (atexit and pytest).

Reply:
I've pushed an update, basically adding this paragraph to the first bullet point:

>   Implementing a better, inspired by Trio nurseries, task spawning API in
  asyncio was the main motivation for this PEP. That work is currently blocked
  on Python not having native language level support for exception groups.

I propose to merge this PR with that, but if you guys want to phrase it differently please open a follow up PR.